components/tabs/impl/brave_tab_strip_collection.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/tabs/public/brave_tab_strip_collection.h"
+
+#include "brave/components/tabs/public/tree_tab_node.h"
+#include "brave/components/tabs/public/tree_tab_node_id.h"
+#include "components/tabs/public/tab_collection.h"
+
+namespace tabs {
+
+#if !BUILDFLAG(IS_ANDROID)
+void BraveTabStripCollection::AddTabRecursive(
+    std::unique_ptr<TabInterface> tab,
+    size_t index,
+    std::optional<tab_groups::TabGroupId> new_group_id,
+    bool new_pinned_state) {
+  if (!in_tree_tab_mode_) {
+    TabStripCollection::AddTabRecursive(std::move(tab), index, new_group_id,
+                                        new_pinned_state);
+    return;
+  }
+
+  // If the previous tab is in the same hierarchy of tree of the opener, we can
+  // add the new tab to the same tree node.
+  if (const auto* opener = tab->GetOpener()) {
+    auto* opener_collection = opener->GetParentCollection(GetPassKey());
+    CHECK_EQ(opener_collection->type(), TabCollection::Type::TREE_NODE);
+
+    CHECK_GE(index, 1u);
+    TabInterface* previous_tab = GetTabAtIndexRecursive(index - 1);
+
+    auto* previous_tab_collection =
+        previous_tab->GetParentCollection(GetPassKey());
+    CHECK_EQ(previous_tab_collection->type(), TabCollection::Type::TREE_NODE);
+
+    if (static_cast<TreeTabNode*>(opener_collection)->GetTopLevelAncestor() ==
+        static_cast<TreeTabNode*>(previous_tab_collection)
+            ->GetTopLevelAncestor()) {
+      auto tree_tab_node = std::make_unique<TreeTabNode>(
+          tree_tab::TreeTabNodeId::GenerateNew(), std::move(tab));
+      opener_collection->AddCollection(std::move(tree_tab_node), 1);
```

Comment:
why wrapped tree node is inserted at `1` always?

Reply:
Because #0 is reserved for the "Tab" for the collection itself. 
i.e. opener collection will have the opener tab at #0

Reply:
IIUC, it seems `previous tab` and `opener tab` are in same hierachy, new tab will be added to right after position of `operner tab` as it's added at `1` always. Am I right?

Reply:
Yes, that's correct.

```
- Opener tab collection(treenode)
   - #0 opener tab interface
   - #1 new tab collection(treenode)
       - #0 new tab interface
   - #2 other tab collection
       - #0 other tab interface
```

Maybe it'd better to add new tabs to the last of child?